pkg/cloud/scope/machine.go

```diff
@@ -386,3 +384,107 @@ func (m *MachineScope) SetInterruptible() {
 		m.AWSMachine.Status.Interruptible = true
 	}
 }
+
```

Comment:
Could you restructure this file more, such that declarations goes up in the file and function goes later?

Reply:
Yes, thinking about it, since this is a straight copy of a dependency we needed, it may be better to keep it into it's own separate file, in case we want to remove it in the future.
Added an extra `providerid.go` file for it.